File: Day246/firstOcuurenceOfX.go
package main

import (
	"fmt"
)

func find(arr []int, n int, x int) []int {
	low, high := 0, n-1
	answer := []int{}

	for low <= high {
		mid := high - ((high - low)>>1)

		if arr[mid] == x {
			for mid >= 1 && arr[mid-1] == x {
				mid--
			}
			answer = append(answer, mid)

			for mid < n-1 && arr[mid+1] == x {
				mid++
			}
			answer = append(answer, mid)
			return answer
		} else if x > arr[mid] {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}

	answer = append(answer, -1)
	answer = append(answer, -1)
	return answer
}

func main() {
	n := 9
	arr := []int{1, 3, 5, 5, 5, 5, 67, 123, 125}
	x := 5
	fmt.Println(find(arr, n, x))
}

File: Day246/firstOcuurenceOfX.java

package Day246;

import java.util.ArrayList;

public class firstOcuurenceOfX {

   public static ArrayList<Integer> find(int arr[], int n, int x)
    {
        // code here
        int low = 0, high = n-1;
        ArrayList<Integer> answer = new ArrayList<>();
        while (low <= high) {
            int mid = high - ((high - low)>>1);
             
            
            if (arr[mid] == x) {
                
                
                while (mid >=1 && arr[mid-1]==x) {
                    mid--;
                }
                answer.add(mid);
                while(mid<n-1 && arr[mid+1]==x) {
                    mid++;
                }
                answer.add(mid);
                return answer;
            }
            else if (x >arr[mid]){
                low = mid + 1;
            } else {
                high = mid -1;
            }
        }
        answer.add(-1);
        answer.add(-1);
        return answer;
    }

    public static void main(String[] args) {
        int n = 9;
        int arr[] = { 1, 3, 5, 5, 5, 5, 67, 123, 125};
        int x = 5;
        System.out.println(find(arr, n , x));
    }
}